src/alloc.rs

```diff
@@ -382,6 +372,58 @@ fn exit_alloc(msg: &str) {
     std::process::exit(1);
 }
 
+// -----------------------------------------------------------------------------
+//        APIs for querying the block information about each allocation
+// -----------------------------------------------------------------------------
+
+pub struct AllocMetadata;
+
+impl AllocMetadata {
+    /// Insert metadata about an allocation block. The GC scans for reachable
+    /// objects conservatively. It uses the metadata stored against each
+    /// allocation to determine:
+    /// 1) the size of an allocation block
+    /// 2) whether an allocation block's memory management is the responsibility
+    ///    of the GC
+    /// 3) whether an arbitrary pointer-like word points within an
+    ///    allocation block
+    pub(crate) fn insert(ptr: usize, size: usize, gc: bool) {
+        ALLOC_LOCK.lock();
+        unsafe {
+            match ALLOC_INFO {
+                Some(ref mut pm) => pm.insert(ptr as usize, size, true),
+                None => {
+                    let mut al = AllocList::new();
+                    al.insert(ptr as usize, size, gc);
+                    ALLOC_INFO = Some(al);
+                }
+            };
+        }
+        ALLOC_LOCK.unlock();
+    }
+    /// Returns metadata about an allocation block when given an arbitrary
```

Comment:
Since this function returns `Option`, I assume you can pass *any* pointer in here? The doc string suggests that only valid pointers can be passed in but I don't think that's correct.

Reply:
Yes, I see what you mean by the wording of the docstring. I'll reword it to make it clearer that you _can_ pass any pointer. But it will only return pointer info in those circumstances.